### PR TITLE
When reading expired sessions - expire them

### PR DESCRIPTION
**Pull request by zeripath**
Update to latest macaron/session following merge of
https://gitea.com/macaron/session/pulls/11

Also remove old memory provider as 11 updates the memory provider to
make it unnecessary.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

**Review by jolheiser — APPROVED** [review]

File: vendor/gitea.com/macaron/session/postgres/postgres.go [hunk 1]
```diff
@@ -121,18 +121,20 @@ func (p *PostgresProvider) Init(maxlifetime int64, connStr string) (err error) {
 
 // Read returns raw session store by session ID.
 func (p *PostgresProvider) Read(sid string) (session.RawStore, error) {
+	now := time.Now().Unix()
 	var data []byte
-	err := p.c.QueryRow("SELECT data FROM session WHERE key=$1", sid).Scan(&data)
+	expiry := now
+	err := p.c.QueryRow("SELECT data FROM session WHERE key=$1", sid).Scan(&data, &expiry)
```
Comment (zeripath) [inline vendor/gitea.com/macaron/session/postgres/postgres.go]: DAMN! This needs to be:

```suggestion
	err := p.c.QueryRow("SELECT data, expiry FROM session WHERE key=$1", sid).Scan(&data, &expiry)
```

Comment (techknowlogick) [reply vendor/gitea.com/macaron/session/postgres/postgres.go]: https://gitea.com/macaron/session/pulls/12 I've used 🧙 powers to merge the linked PR, you should be able to use a go get -u on sessions and a make vendor

Comment (zeripath) [reply vendor/gitea.com/macaron/session/postgres/postgres.go]: I'm having difficulty getting go get -u to update it...

Comment (zeripath) [reply vendor/gitea.com/macaron/session/postgres/postgres.go]: done

**Pull request merged**